x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/DateTimeToCharProcessorTests.java

```diff
@@ -0,0 +1,298 @@
+                );
+            }).collect(Collectors.joining("\n"));
+        }
+    
+        /**
+         * Saves the generated test script into the file specified as the first and only argument.
+         *
+         * Once the file is generated, you can execute the following command to generate the output dataset with:
+         *
+         * <p>
+         *  <code>
+         *  PGPASSWORD="mysecretpassword" psql --quiet -h localhost -p 5432 -U postgres -f /tmp/postgresql-tochar-test.sql &gt; /tochar.csv
+         *  </code>
+         * </p>
+         */
+        public static void main(String[] args) throws Exception {
+            String scriptFilename = args.length < 1 ? "/tmp/postgresql-tochar-test.sql" : args[0];
+            Files.writeString(Path.of(scriptFilename), unitTestExporterScript(), StandardCharsets.UTF_8);
+        }
+    }
+
+    @ParametersFactory(argumentFormatting = "%1$s: timestamp=%2$s, zone=%3$s, format=%4$s")
+    public static List<Object[]> readCsv() throws Exception {
+        Path testFilePath = Path.of(DateTimeToCharProcessorTests.class.getResource("tochar.csv").toURI());
+        return Files.readAllLines(testFilePath).stream().map(line -> {
+            String[] cols = line.split(Pattern.quote(TestGenerator.DELIMITER));
+            return new Object[]{ cols[0], cols[1], cols[2], cols[3], cols[4] };
+        }).collect(Collectors.toList());
+    }
+
+    private final String id;
+    private final String secondsAndFractionsSinceEpoch;
+    private final String zone;
+    private final String formatString;
+    private final String expectedResult;
+    
+    private Locale defaultLocale = null;
+
+    public DateTimeToCharProcessorTests(String id, String secondsAndFractionsSinceEpoch, String zone, String formatString, 
+        String expectedResult) {
+        this.id = id;
+        this.secondsAndFractionsSinceEpoch = secondsAndFractionsSinceEpoch;
+        this.zone = zone;
+        this.formatString = formatString;
+        this.expectedResult = expectedResult;
+    }
+
+    private static ZonedDateTime dateTimeWithFractions(String secondAndFractionsSinceEpoch) {
+        BigDecimal b = new BigDecimal(secondAndFractionsSinceEpoch);
+        long seconds = b.longValue();
+        int fractions = b.remainder(BigDecimal.ONE).movePointRight(9).intValueExact();
+        int adjustment = 0;
+        if (fractions < 0) {
+            fractions += 1e9;
+            adjustment = -1;
+        }
+        return dateTime((seconds + adjustment) * 1000).withNano(fractions);
+    }
+
+    /**
+     * Since the test dataset was exported from PostgreSQL using en_US locale, 
+     * let's stick with to that Locale during the testing of this function.
+     */
+    @Before
+    public void changeLocaleToUS() {
+        this.defaultLocale = Locale.getDefault();
+        Locale.setDefault(Locale.US);
+    }
+    
+    @After
+    public void revertLocale() {
+        Locale.setDefault(this.defaultLocale);
+    }
```

Comment:
I think our tests don't run in parallel in the same JVM, so this should be safe, but if you have a better idea how to do this, I am all ears.

Reply:
After adding the `Locale.ROOT` to the `String.format` and `String.to{Lower,Upper}Case`, this can be removed. Yay!